### PR TITLE
Add necessary route handlers

### PR DESCRIPTION
**Pull request by kylegach**
1. `/versions.json`
2. `/whats-new/v1/`
3. `/event-log`

I also went ahead and added a proxy redirect for `/day/*`

### CODE REVIEW

File: apps/frontpage/app/versions/route.ts [hunk 1]
```diff
@@ -1,44 +1,33 @@
-import { stringify } from 'node:querystring';
```
Comment (kylegach) [inline apps/frontpage/app/versions/route.ts]: The left side is a direct copy/paste of https://github.com/storybookjs/frontpage/blob/main/src/functions/versions.js, for easy diffing.

File: apps/frontpage/app/versions/route.ts [hunk 2]
```diff
@@ -1,44 +1,33 @@
...
+import type { NextRequest } from 'next/server';
 import { BigQuery } from '@google-cloud/bigquery';
-import semver from 'semver';
-
-const CACHE_KEY = 'versions';
-const cache = new LRU({
-  max: 64 * 1024, // 64kb
-  maxAge: 10 * 60 * 60 * 1000, // 10 mins
-  length: (n, key) => n.length,
-});
```
Comment (kylegach) [inline apps/frontpage/app/versions/route.ts]: According to [Next.js docs](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#caching), caching is built in to route handlers like this one.

File: apps/frontpage/next.config.js [hunk 3]
```diff
@@ -53,6 +53,11 @@ module.exports = {
         destination: '/docs/configure/telemetry',
         permanent: true,
       },
+      {
+        source: '/versions.json',
```
Comment (kylegach) [inline apps/frontpage/next.config.js:57]: `versions.json` was not a valid route name, so I had to configure this redirect. The query parameters are forwarded, so everything still works as it should.

**Review by cdedreuille — APPROVED** [review]
This looks great !! Thanks for working on this @kylegach

**Pull request merged**